stubs/inertia/resources/js/Pages/Auth/TwoFactorChallenge.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <jet-authentication-card>
+        <template #logo>
+            <jet-authentication-card-logo />
+        </template>
+
+        <div class="mb-4 text-sm text-gray-600">
+            <template v-if="!recovery">
+                Please confirm access to your account by entering the authentication code provided by your authenticator application.
+            </template>
+            <template v-else>
+                Please confirm access to your account by entering one of your emergency recovery codes.
+            </template>
+        </div>
+
+        <jet-validation-errors class="mb-4" />
+
+        <form @submit.prevent="submit">
+            <div v-if="!recovery">
+                <jet-label for="code" value="Code" />
+                <jet-input ref="code" id="code" type="text" class="mt-1 block w-full" v-model="form.code" autofocus autocomplete="one-time-code" />
+            </div>
+            <div v-else>
+                <jet-label for="recovery_code" value="Recovery Code" />
+                <jet-input ref="recovery_code" id="recovery_code" type="text" class="mt-1 block w-full" v-model="form.recovery_code" autocomplete="one-time-code" />
+            </div>
+
+            <div class="flex items-center justify-end mt-4">
+                <button @click.prevent="toggleRecovery" class="text-sm text-gray-600 hover:text-gray-900 underline cursor-pointer">
+                    <template v-if="!recovery">
+                        Use a recovery code
+                    </template>
+                    <template v-else>
+                        Use an authentication code
+                    </template>
+                </button>
+                <jet-button class="ml-4" :class="{ 'opacity-25': form.processing }" :disabled="form.processing">
+                    Login
+                </jet-button>
+            </div>
+        </form>
+    </jet-authentication-card>
+</template>
+<script>
+    import JetAuthenticationCard from '../../Jetstream/AuthenticationCard'
+    import JetAuthenticationCardLogo from '../../Jetstream/AuthenticationCardLogo'
+    import JetButton from '../../Jetstream/Button'
+    import JetInput from './../../Jetstream/Input'
+    import JetLabel from './../../Jetstream/Label'
+    import JetValidationErrors from '../../Jetstream/ValidationErrors'
+
+    export default {
+        components: {
+            JetAuthenticationCard,
+            JetAuthenticationCardLogo,
+            JetButton,
+            JetInput,
+            JetLabel,
+            JetValidationErrors,
+        },
+        data() {
+            return {
+                recovery: false,
+                form: this.$inertia.form({
+                    code: '',
+                    recovery_code: '',
+                })
+            }
+        },
+        methods: {
+            toggleRecovery() {
+                this.recovery ^= true
```

Comment:
```suggestion
                this.recovery = true
```

Reply:
They are using this to toggle between 0 and 1.

Reply:
Wouldn't it be less confusing to use a boolean?

Reply:
This is actually correct. It's a bitwise XOR that changes true to false, and false to true in this situation (hence toggle) 👍

Reply:
That's actually a neat trick!

Reply:
For sure :) Also works in PHP ;)